cmd/object-handlers.go

```diff
@@ -1076,6 +1076,9 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 			return
 		}
+		if globalIsGateway && globalGatewayName == "s3" && objTags == "" {
+			srcInfo.UserDefined[xhttp.AmzObjectTagging] = objTags
```

Comment:
do you need this if you are already handling this inside s3 gateway implementation?

Reply:
`objTags` are only entered in `srcInfo.UserDefined` if they are not empty. We only need the empty tags in `srcInfo.UserDefined` for the specific case of tagging-directive being REPLACE with no tags given, which is why we are making the check. If tags are not in `srcInfo.UserDefined` the tagging operation is assumed to be the default - COPY.

Reply:
This a bit of a heavy handed check, is it not possible to do without this check? why not just do this with 

```
if globalIsGateway && objTags == "" {
```

Other gateway implementations handle the conversion of x-amz-meta back into S3 compatible form

Reply:
If check for `s3` is not required, then the correct fix is simply to do
```
if globalIsGateway {
    srcInfo.UserDefined[xhttp.AmzTagDirective] = "REPLACE"
}
```

Reply:
Harsha, if Tags are being sent as part of Put or copy object, should it not fail if it is not `S3` Gateway

Reply:
@kannappanr if we preserve them in UserDefined then it can be supported with all backends